design/design.md.md

```diff
@@ -0,0 +1,114 @@
+Please find the terminology below. Basic format for "For" Loops is given below
+
+For (iterator in Values) 
+    block of statements 
+
+For this week, we are only focusing on cases where "Values" in the above format is derived from one builtin function (Range and its variants)
+
+
+TestCase 1: Validate that "Values" in the loop cannot be Range(). Throw a typecheck error for numer of arguments
+
+Input:
+
+i:int = 0
+for i in range():
+  print(i)
+
+Expected Output : TYPE ERROR: expected 2 parameters instead got 0
+
+TestCase 2: Parameters of range should be integers and start <=end. Else, throw typecheck error (design decision)
+
+Input
+i:int = 0
+for i in range(10,1):
```

Comment:
Some of these range examples have 2 arguments and some only 1. How will that work?

Reply:
We were planning to implement basic function overloading (using different names for the same function based on the parameters as of now) until the code from other teams is available.